doc/addon.md

```diff
@@ -8,20 +10,20 @@ variables because doing so does not take into account the fact that an add-on
 may be loaded into multiple threads nor that an add-on may be loaded multiple
 times into a single thread.
 
-The `Napi::Addon` class can be used to define an entire add-on. Instances of
-`Napi::Addon` subclasses become instances of the add-on, stored safely by
+The `Napi::Addon<T>` class can be used to define an entire add-on. Instances of
+`Napi::Addon<T>` subclasses become instances of the add-on, stored safely by
```

Comment:
This is only available in the recent versions right. I'm wondering if we needed some versioning so that people can tell when new APIs are available.